CMakeLists.txt

```diff
@@ -779,6 +781,17 @@ if(swift_build_windows AND NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
   configure_sdk_windows("Windows" "msvc" "${SWIFT_SDK_WINDOWS_ARCHITECTURES}")
 endif()
 
+is_sdk_requested(WASI swift_build_wasi)
+if(swift_build_wasi AND NOT "${SWIFT_HOST_VARIANT_SDK}" STREQUAL "WASI")
+  if(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY OR SWIFT_BUILD_DYNAMIC_STDLIB)
+    message(FATAL_ERROR "Unable to build dynamic overlay/stdlib for WASI. WASM does not support dynamic linking.")
+  endif()
```

Comment:
This doesn't really work this way.  This setup needs to permit dynamic linking.  The problem is that you can build multi-target currently.  If you build on Windows, you cannot build the static version, so this breaks.  This is one of the motivation behind my recent work to rework the build system to separate out the SDK builds.

Reply:
I see. OK, I'll drop this validation.

Reply:
Do you have a link to any PRs about that, @compnerd ?

I was looking at `CMAKE_AR` recently, because WASM needs to use `llvm-ar`, but I couldn't find a way to set that for a particular target (I think clang will look for it in the path given with the `-B` flag, but not sure). Do we just rely on luck that the given archiver won't return an error for all target platforms?

Reply:
The archiver is not invoked by clang, it is invoked by ninja as instructed by CMake.  CMAKE_AR is the right tool to change the archiver.

The last series of my changes on the build system are related work.  They are generally beneficial, but, are required for me to make progress on the WASM/WASI port.  Anything other than Darwin will be benefiting from that work.